import "galaxy.proto";
package baidu.galaxy.proto;

option cc_generic_services = true;

message SubmitJobRequest {

}
message SubmitJobResponse {

}
message UpdateJobRequest {

}
message UpdateJobResponse {

}
message RemoveJobRequest {

}
message RemoveJobResponse {

}

message ListJobsRequest {
} 

message ListJobsResponse {
}

message ShowJobRequest {
}

message ShowJobResponse {
}

message AssignTaskRequest {

}
message AssignTaskResponse {

}
message ListTasksRequest {

}
message ListTasksResponse {

}

message UpdateTaskRequest {

}
message UpdateTaskResponse {

}
message ExecuteCmdRequest {

}
message ExecuteCmdResponse {

}


message FectTaskRequest {
}

message FetchTaskResponse {
}


message StopJobRequest {} 
message StopJobResponse {}

service AppMaster {
    //for app-worker: report & task assignment
    rpc FetchTask(FectTaskRequest) returns (FetchTaskResponse);

   // rpc Status(StatusRequest) returns (StatusResponse);

    //job man
    rpc SubmitJob(SubmitJobRequest) returns (SubmitJobResponse);
    rpc UpdateJob(UpdateJobRequest) returns (UpdateJobResponse);

    rpc StopJob(StopJobRequest) returns (StopJobResponse);
    rpc RemoveJob(RemoveJobRequest) returns (RemoveJobResponse);
    rpc ListJobs(ListJobsRequest)   returns (ListJobsResponse);
    rpc ShowJob(ShowJobRequest) returns (ShowJobResponse);

    //[optional]
    rpc ExecuteCmd(ExecuteCmdRequest) returns (ExecuteCmdResponse); 
    // op
}
